src/simmate/apps/clease/workflows/cluster_expansion.py

```diff
@@ -339,14 +353,87 @@ def run_config(
         # composition + MC run. Then I'd put all results together in a
         # pandas dataframe and write to a CSV. Maybe write a new CSV each cycle
         # too.
 
         # TODO: write summary files using our observers
-        # energy_progression = observer_01.energies
-        # structure_best = observer_02.atoms
+        energy_progression = observer_01.energies
+        with open("energy_progression.csv", "a", newline="") as f:
+            writer = csv.writer(f)
+            writer.writerow(energy_progression)
+        for row in observer_02:
+            structure_best = row.atoms
+            chemform = structure_best.get_chemical_formula()
+            structure_best.write(chemform, format="cif")
 
         # TODO: generate and plot the hull diagram using pymatgen
+        # CLEASE currently outputs the concentration of the changing ion from MC
+        # Need to pull composition to use pymatgen phase diagram generator
+        # Need to incorporate multiple cluster expansion models in most cases
+        # Therefore, the phase diagram creation might need to be done manually
+        # from pymatgen.analysis import phase_diagram # import PDEntry, PhaseDiagram
+        # import pandas as pd
+        import csv
+
+        row = (
+            "Energy",
+            "Concentration without transformation",
+            "Concentration  with transformation",
+        )
+        with open("thermo_quantities.csv", "r") as readFile:
+            rd = csv.reader(readFile)
+            lines = list(rd)
+            lines.insert(0, row)
+
+        with open("thermo_quantities.csv", "w", newline="") as writeFile:
+            wt = csv.writer(writeFile)
+            wt.writerows(lines)
+
+        readFile.close()
+        writeFile.close()
+        """
+        file = pd.read_csv("thermo_quantities.csv")
+        #comp = pd.read_csv("O3_comp_neweci.xlsx")
+        dictionary = file.to_dict('records')
+        lis = []
+        for x in dictionary:
+            new_dict = {k:v for k,v in x.items() if v != 0}
+            lis.append(new_dict)
+        data = file['Energy']
+        num = 0
+        entries = []
+        for y in lis:
+            dis = data.iloc[num]
+            entries.append(phase_diagram.PDEntry(composition=y, energy=dis))
+            num += 1
+    
+        phase = phase_diagram.PhaseDiagram(entries=entries)
+        new_list = []
+        #for all points
+        for entry in entries:
+            energy = phase.get_form_energy_per_atom(entry)
+            dicti = {entry.composition:energy}
+            new_list.append(dicti)
+
+        #energy above hull
+        for entry in entries:
+            energy = phase.get_e_above_hull(entry)
+            dicti = {entry.composition:energy}
+            new_list.append(dicti)
+
+        with open("O3_hull_neweci.csv", "w", newline='') as f:
+            write = csv.writer(f)
+            for a in new_list:
+                write.writerow(a.values())
+        """
```

Comment:
just a tip -- ctrl+1 in spyder comments out sections. Triple quote comments are normally just for writing user documentation